src/main/java/net/neoforged/neoforge/network/registration/ModdedPacketRegistrar.java

```diff
@@ -56,15 +57,15 @@ public Map<ResourceLocation, PlayRegistration<?>> getPlayRegistrations() {
     public <T extends CustomPacketPayload> IPayloadRegistrar play(ResourceLocation id, FriendlyByteBuf.Reader<T> reader, IPlayPayloadHandler<T> handler) {
         play(
                 id, new PlayRegistration<>(
-                        reader, handler, version, Optional.empty(), optional));
+                        reader, handler, ModLoadingContext.get().getActiveNamespace(), version, Optional.empty(), optional));
```

Comment:
New code should not be using the static `ModLoadingContext` - see https://github.com/neoforged/FancyModLoader/issues/39

Use the Namespace of the provided id for this (and all other instances).

Reply:
See other comments as we're going back and forth. :P

Reply:
Hello, thank you for your review! I also originally considered not using the `ModLoadingContext`. Instead, I proposed changing the way the currently existing `modId` field of `ModdedPacketRegistrar` is handled, in order to no longer restrict it to needing to match the namespaces of all channels registered through that registrar, so that I could use it for accessing the id of the mod that registers the channels via the given registrar, but I was told to use the `ModLoadingContext` to retrieve the current mod id as it requires less changes. (For more details and also other reasons why the mentioned `modId` field should be refactored like this, see [this](https://discord.com/channels/313125603924639766/852298000042164244/1214236463584247819) and the following related messages.)